osu.Game/Screens/Edit/EditorTable.cs

```diff
@@ -27,6 +27,8 @@ public abstract partial class EditorTable : TableContainer
 
         public const int TEXT_SIZE = 14;
 
+        public static bool locked_table = false;
+
```

Comment:
Sorry for the late reply. I made it static because there would be only one instance at a time per user and made it easy to access in `RowBackground`. I can change It though to fit Dependency Injection It's just when you say the table is cached do you mean this initialization of `ControlPointTable`?
https://github.com/ppy/osu/blob/7aeab174eb0ec8711091ea4d99906d858212538d/osu.Game/Screens/Edit/Timing/ControlPointList.cs#L24
Cause to my understanding it's not cached for injection so I wouldn't be able to pull it with resolved. I could cache it though and It would work as I think you've implied. Though If I'm wrong please do point it out.

Reply:
Forget what I said about DI. It's just that I have mixed feelings with having a static variable to pass information to other components.

Reply:
It's not only you. We don't use static variables in this project.